src/tools/crawler/network.rs

```diff
@@ -11,6 +11,15 @@ use ziggurat_core_crawler::connection::KnownConnection;
 /// The elapsed time before a connection should be regarded as inactive.
 pub const LAST_SEEN_CUTOFF: u64 = 10 * 60;
 
+#[derive(Debug, Default, Clone, Copy, PartialEq)]
+pub enum NodeState {
```

Comment:
maybe it's a bit more general approach (so it could be reused even, but we won't need it anywhere else here though):
```suggestion
pub enum ConnectionState {
```

Reply:
You're right - changed.